lib/SIL/IR/SILDeclRef.cpp

```diff
@@ -376,11 +376,8 @@ SILLinkage SILDeclRef::getLinkage(ForDefinition_t forDefinition) const {
     if (isSerialized())
       return maybeAddExternal(SILLinkage::PublicNonABI);
 
-    d = cast<NominalTypeDecl>(d->getDeclContext());
+    limit = Limit::NeverPublic;
 
-    // FIXME: This should always be true.
-    if (d->getModuleContext()->isResilient())
```

Comment:
Let's convert this into an assertion to be sure. There's probably a way to get the module context of the original `d` that doesn't require it to be a NominalTypeDecl?